verified_boot.md

```diff
@@ -0,0 +1,240 @@
+- Intel specific, AMD has a different but comparable secure boot scheme
+  Verifies the initial bootblock, rest is up to BIOS (Vboot)
+- fused: programmed into one-time programmable e-fuses inside the chipset
+- Measurements are made in locality 3, so they can't be faked by malicious BIOS
+- Profile 3: Verified and Measured with infinite time for remediation
+- Profile 3 acts as root of trust for measurement, but not for verification
+- On verification failure, the PCR0 measurements are always the same. This was
+  determined experimetally.
+- so users can choose to use Dasharo firmware and get measured IBB, or flash
+  their own and lose measured IBB.
+  - might make sense for some small percentage for users
+  - does not help us a lot
+
+---
+
+# Boot Guard cons
+
+- Requires more blobs in firmware
+  - Boot Guard ACM
+- Fusing removes some owner control
+- Profile 3 does not ensure initial boot block authenticity
+  - Only helps us establish a root of trust for measurement
+- Other profiles take away owner control completely
+
+???
+
+- ACM - Authenticated Code Module. Hyper-privileged, Intel signed blob that
+  performs the actual BIOS verification and measurement, then hands off to the
+  BIOS reset vector.
+
+---
+
+# CBFS Verification
+
+- A relatively new coreboot feature
+- Cryptographically verifies components of the coreboot image
+- According to documentation:
+  > This only makes sense if you use some
+  > out-of-band mechanism to guarantee the integrity of the bootblock
+  > itself, such as Intel Boot Guard or flash write-protection.
+- Depends on some other mechanism for signing (e.g. Boot Guard)
+- Can effectively replace the portions of Vboot that Dasharo uses
+
+???
+
+- CBFS - coreboot's internal file system
+
+---
+
+# Firmware updates
+
+- Chrome Vboot: A+B+RO
+  - A and B slots for updates, RO slot contains IBB and verification code
+  - Anti-rollback using TPM
+  - Vboot aware OS service manages slot updates
+- Dasharo:
+  - Initial idea was to leave RO untouched and only update A/B slots
+  - In practice, most updates introduce breaking changes that require updating
+    the bootblock
+  - Dasharo Tools Suite handles updates, does the entire update in one operation
+    - So we don't use the A/B feature
+  - Capsule Updates will help here
+
+---
+
+# Other firmware update mechanisms
+
+- Flashrom plugin in fwupd
+  - Updates the BIOS region only
+  - Is not Vboot aware
+- UEFI Capsule Update
+  - Most widely used in proprietary UEFIs
+  - Requires an update handler in SMM
+  - Can be made aware of which regions to update and which to preserve, verify
+    signatures, handle disabling flash protections
+  - WIP for Dasharo
+
+???
+
+- Capsule Updates are a requirement for Windows sticker certification
+- Supported by Windows Update and fwupd
+
+# Other firmware update mechanisms
+
+- Intel BIOS Guard
+  - Replaces the SMM capsule update handler with an ACM
+  - Proprietary tooling
+  - Can bypass chipset flash protections
```

Comment:
Only if such option is selected in MFIT/FITC/FIT on the ME side.

Reply:
https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR222-R223